thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/onboard/tasks/FunctionCreationOnboardingTask.java

```diff
@@ -0,0 +1,193 @@
+  public static final String ALERT_ID = DefaultDetectionOnboardJob.ALERT_ID;
+  public static final String ALERT_NAME = DefaultDetectionOnboardJob.ALERT_NAME;
+  public static final String ALERT_CRON = DefaultDetectionOnboardJob.ALERT_CRON;
+  public static final String ALERT_FROM = DefaultDetectionOnboardJob.ALERT_FROM;
+  public static final String ALERT_TO = DefaultDetectionOnboardJob.ALERT_TO;
+  public static final String ALERT_APPLICATION = DefaultDetectionOnboardJob.ALERT_APPLICATION;
+  public static final String DEFAULT_ALERT_RECEIVER = DefaultDetectionOnboardJob.DEFAULT_ALERT_RECEIVER_ADDRESS;
+
+  public static final Boolean DEFAULT_IS_ACTIVE = true;
+  public static final String DEFAULT_METRIC_FUNCTION = "SUM";
+  public static final String DEFAULT_WINDOW_DELAY = "0";
+  public static final String DEFAULT_ALERT_CRON = "0 0/5 * 1/1 * ? *"; // Every 5 min
+
+  private AnomalyFunctionManager anoomalyFunctionDAO;
+  private AlertConfigManager alertConfigDAO;
+  private DatasetConfigManager datasetConfigDAO;
+
+  public FunctionCreationOnboardingTask() {
+    super(TASK_NAME);
+    DAORegistry daoRegistry = DAORegistry.getInstance();
+    this.alertConfigDAO = daoRegistry.getAlertConfigDAO();
+    this.anoomalyFunctionDAO = daoRegistry.getAnomalyFunctionDAO();
+    this.datasetConfigDAO = daoRegistry.getDatasetConfigDAO();
+  }
+
+  /**
+   * Executes the task. To fail this task, throw exceptions. The job executor will catch the exception and store
+   * it in the message in the execution status of this task.
+   */
+  @Override
+  public void run() {
+    DetectionOnboardTaskContext taskContext = getTaskContext();
+    DetectionOnboardExecutionContext executionContext = taskContext.getExecutionContext();
+    Configuration configuration = taskContext.getConfiguration();
+
+    Preconditions.checkNotNull(executionContext.getExecutionResult(FUNCTION_FACTORY));
+    Preconditions.checkNotNull(executionContext.getExecutionResult(ALERT_FILTER_FACTORY));
+
+    AnomalyFunctionFactory anomalyFunctionFactory = (AnomalyFunctionFactory)
+        executionContext.getExecutionResult(FUNCTION_FACTORY);
+    AlertFilterFactory alertFilterFactory = (AlertFilterFactory) executionContext.getExecutionResult(ALERT_FILTER_FACTORY);
+
+    Preconditions.checkNotNull(anomalyFunctionFactory);
+    Preconditions.checkNotNull(alertFilterFactory);
+
+    // Assert if null
+    Preconditions.checkNotNull(taskContext);
+    Preconditions.checkNotNull(configuration.getString(FUNCTION_NAME));
+    Preconditions.checkNotNull(configuration.getString(COLLECTION_NAME));
+    Preconditions.checkNotNull(configuration.getString(METRIC_NAME));
+    Preconditions.checkNotNull(configuration.getString(CRON_EXPRESSION));
+    Preconditions.checkNotNull(configuration.getString(WINDOW_SIZE));
+    Preconditions.checkNotNull(configuration.getString(WINDOW_UNIT));
+    Preconditions.checkNotNull(configuration.getString(PROPERTIES));
```

Comment:
Hi YungYu, if here we assert the input function properties should be not null, where do we generate the default function properties? Thought when creating functions, we are using the default values stored in anomaly function already.